### PR TITLE
Merge release/1.0.0 into master

### PR DESCRIPTION
**Pull request by eerhardt**
We had quite a few changes go into release/1.0.0 that didn't go into master, but should have.  In order to ensure master contains all the changes, I did a 'mop-up' RI.

There isn't that many files changed, but it is really easy to make a mistake.  Can everyone check out the changes and ensure they are meant to be in master?

@ericstj @weshaggard @brthor @schellap @mellinoe

### CODE REVIEW

**Comment by eerhardt** [comment]
test CentOS7.1 x64 Debug Build

**Comment by eerhardt** [comment]
Can someone take a look at this to double check these are the right changes for the 'master' branch?  I'd like to get this merged today.

File: build_projects/dotnet-host-build/CompileTargets.cs [hunk 1]
```diff
@@ -267,9 +267,9 @@ public static BuildTargetResult GenerateMSbuildPropsFile(BuildTargetContext c)
             msbuildProps.AppendLine($"    <HostVersion>{hostVersion.LatestHostPolicyVersion.WithoutSuffix}</HostVersion>");
             msbuildProps.AppendLine($"    <HostResolverVersion>{hostVersion.LatestHostFxrVersion.WithoutSuffix}</HostResolverVersion>");
             msbuildProps.AppendLine($"    <HostPolicyVersion>{hostVersion.LatestHostVersion.WithoutSuffix}</HostPolicyVersion>");
-            msbuildProps.AppendLine($"    <BuildNumberMajor></BuildNumberMajor>");
-            msbuildProps.AppendLine($"    <BuildNumberMinor></BuildNumberMinor>");
-            msbuildProps.AppendLine($"    <PreReleaseLabel></PreReleaseLabel>");
+            msbuildProps.AppendLine($"    <BuildNumberMajor>{hostVersion.LatestHostBuildMajor}</BuildNumberMajor>");
+            msbuildProps.AppendLine($"    <BuildNumberMinor>{hostVersion.LatestHostBuildMinor}</BuildNumberMinor>");
```
Comment (brthor) [inline build_projects/dotnet-host-build/CompileTargets.cs]: Are these versions right? HostVersion is already at `1.0.2`, is that what we want here?

Comment (eerhardt) [inline build_projects/dotnet-host-build/CompileTargets.cs]: @schellap - can you answer that?

FYI - this is what is currently in 'master':  https://github.com/dotnet/core-setup/blob/master/build_projects/dotnet-host-build/CompileTargets.cs#L263-L274


This diff was me fixing up the bad merge from the previous merge commit.

File: build_projects/shared-build-targets-utils/Utils/HostVersion.cs [hunk 2]
```diff
@@ -64,9 +64,9 @@ public override string ToString()
         public string LatestHostBuildMajor => CommitCountString;
         public string LatestHostBuildMinor => "00";
         public bool EnsureStableVersion => false;
-        public VerInfo LatestHostVersion => new VerInfo(1, 0, 2, ReleaseSuffix, LatestHostBuildMajor, LatestHostBuildMinor);
-        public VerInfo LatestHostFxrVersion => new VerInfo(1, 0, 2, ReleaseSuffix, LatestHostBuildMajor, LatestHostBuildMinor);
-        public VerInfo LatestHostPolicyVersion => new VerInfo(1, 0, 2, ReleaseSuffix, LatestHostBuildMajor, LatestHostBuildMinor);
+        public VerInfo LatestHostVersion => new VerInfo(1, 0, 2, ReleaseSuffix, LatestHostBuildMajor, LatestHostBuildMinor, CommitCountString);
```
Comment (brthor) [inline build_projects/shared-build-targets-utils/Utils/HostVersion.cs]: This constructor is pretty confusing. I don't understand why we have commit count twice, but don't block the merge on it.

File: build_projects/dotnet-host-build/CompileTargets.cs (hunk 1 above)
Comment (weshaggard) [inline build_projects/dotnet-host-build/CompileTargets.cs]: This looks correct to me.

Comment (schellap) [inline build_projects/dotnet-host-build/CompileTargets.cs]: Hostversion and hostpolicy version is swapped out.

Comment (schellap) [inline build_projects/dotnet-host-build/CompileTargets.cs]: @brthor, The suffix, build major and minor is what netcore.app uses. It has its own versioning in the pkgproj.

Comment (brthor) [inline build_projects/dotnet-host-build/CompileTargets.cs]: okay

File: build_projects/dotnet-host-build/CompileTargets.cs [hunk 3]
```diff
@@ -267,9 +267,9 @@ public static BuildTargetResult GenerateMSbuildPropsFile(BuildTargetContext c)
             msbuildProps.AppendLine($"    <HostVersion>{hostVersion.LatestHostPolicyVersion.WithoutSuffix}</HostVersion>");
             msbuildProps.AppendLine($"    <HostResolverVersion>{hostVersion.LatestHostFxrVersion.WithoutSuffix}</HostResolverVersion>");
             msbuildProps.AppendLine($"    <HostPolicyVersion>{hostVersion.LatestHostVersion.WithoutSuffix}</HostPolicyVersion>");
```
Comment (brthor) [inline build_projects/dotnet-host-build/CompileTargets.cs]: referrring to here @schellap ?

Comment (schellap) [inline build_projects/dotnet-host-build/CompileTargets.cs]: Yes, this should be swapped.

Should be:

`HostVersion>{hostVersion.LatestHostVersion.WithoutSuffix}</HostVersion>"`

Comment (eerhardt) [inline build_projects/dotnet-host-build/CompileTargets.cs]: That is how it is currently in 'master'.  Can you open a different PR/issue for this?  I'd like to not conflate the merge.

**Comment by brthor** [comment]
let's fix the versions after we merge, it won't break anything now anyways :shipit:

**Pull request merged**